omnifit/spectrum/spectrum.py

```diff
@@ -4,765 +4,840 @@
+
+    @clonable
+    def convert2(self, newunit):
+        """
+        convert2(newunit, clone=False)
+
+        Convert the x axis data to given spectral units.
+        Re-sort the data afterwards.
+
+        Parameters
+        ----------
+        newunit : `astropy.units.core.Unit`
+            Desired (spectral) unit the x axis data should be
+            converted to.
+        clone : `bool`, optional
+            If set to True, returns a modified copy of the spectrum instead
+            of operating on the existing spectrum.
+        """
+        with u.set_enabled_equivalencies(u.equivalencies.spectral()):
+            self.x = self.x.to(newunit)
+        self.__sort()
+
+    @clonable
+    def subspectrum(self, limit_lower, limit_upper):
+        """
+        subspectrum(limit_lower, limit_upper, clone=False)
+
+        Cropped the spectrum along along the x axis using the given
+        inclusive limits.
+
+        Parameters
+        ----------
+        limit_lower : `float`
+            The desired minimum x axis of the cropped spectrum, in
+            current units of the spectrum. This limit is inclusive.
+        limit_upper : `float`
+            The desired maximum x axis of the cropped spectrum, in
+            current units of the spectrum. This limit is inclusive.
+        clone : `bool`, optional
+            If set to True, returns a modified copy of the spectrum instead
+            of operating on the existing spectrum.
+        """
+        iSub = np.logical_and(
+            np.greater_equal(self.x.value, limit_lower),
+            np.less_equal(self.x.value, limit_upper))
+        newX = self.x[iSub]
+        newY = self.y[iSub]
+        self.x = newX
+        self.y = newY
+
+    @clonable
+    def interpolate(self, target_spectrum):
+        """
+        interpolate(target_spectrum,clone=False)
+
+        Interpolate spectrum to match target spectrum resolution.
+        Does not modify current spectrum, but returns a new one, which is
+        a copy of the current spectrum but with the interpolated data on
+        the x and y fields.
+        The target spectrum has to be using the same units on the x and
+        y axes as the current spectrum, or the interpolation will fail.
+
+        Parameters
+        ----------
+        target_spectrum : `BaseSpectrum`
+            The target spectrum which the x axis resolution of the current
+            spectrum should be made to match.
+        clone : `bool`, optional
+            If set to True, returns a modified copy of the spectrum instead
+            of operating on the existing spectrum.
+        """
+        if self.x.unit != target_spectrum.x.unit:
+            raise u.UnitsError('Spectrums have different units on x axis!')
+        if self.y.unit != target_spectrum.y.unit:
+            raise u.UnitsError('Spectrums have different units on y axis!')
+        newX = target_spectrum.x
+        newY = np.interp(newX, self.x, self.y)*self.y.unit
```

Comment:
```suggestion
        if not self.x.unit.is_equivalent(target_spectrum.x.unit, equivalencies=u.spectral()):
            raise u.UnitsError('Spectra have incompatible units on x axis!')
        if not self.y.unit.is_equivalent(target_spectrum.y.unit, equivalencies=u.spectral_density(self.x)):
            raise u.UnitsError('Spectra have incompatible units on y axis!')
        newX = target_spectrum.x
        newY = np.interp(newX, self.x, self.y)
```
See my comment on #38 – recent Astropy+Numpy already conserve units.
The `y` axes units check assumes they may also contain spectral flux density-like data.

Reply:
Fixes implemented in 275c50f ; thank you :)